app/models/git_repository.rb

```diff
@@ -29,14 +29,16 @@ def update!(executor: TerminalExecutor.new(StringIO.new))
     executor.execute!("cd #{repo_cache_dir}", 'git fetch -p')
   end
 
-  def commit_from_ref(git_reference)
-    description = Dir.chdir(repo_cache_dir) do
-      IO.popen(['git', 'describe', '--long', '--tags', '--all', git_reference]) do |io|
+  def commit_from_ref(git_reference, length: 7)
+    Dir.chdir(repo_cache_dir) do
```

Comment:
maybe end the chdir block right here and then deal with the description outside of it, might make refactoring into a helper more obvious